core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -3187,6 +3190,32 @@ private void analyzeSelectAllColumns(
             }
         }
 
+        private List<Field> filterInaccessibleFields(List<Field> fields)
+        {
+            if (!analysis.isHideInaccesibleColumns()) {
+                return fields;
+            }
+            //collect fields by table
+            ListMultimap<QualifiedObjectName, Field> tableFieldsMap = Multimaps.newListMultimap(new HashMap<QualifiedObjectName, Collection<Field>>(), () -> new LinkedList<>());
```

Comment:
Why is this that complex?

As I understand these fields always come from the same relation (maybe you do `checkArgument` to assert that), so do we need to use multimap here?

Reply:
In case of join query like:
`SELECT * FROM nation,customer WHERE customer.nationkey = nation.nationkey AND nation.name = 'FRANCE' AND customer.name='Customer#000001477'`
I receive here fields from multiple tables

and in case of queries like:
`SELECT * FROM (select 'test')`
`SELECT * FROM (SELECT concat(name,'-test') FROM nation WHERE name = 'FRANCE')`

I receive fields without OriginaleTable/OriginaleColumn
so I must manage also those cases.

I'll add tests for that.